go-controller/hybrid-overlay/pkg/controller/overlay_windows.go

```diff
@@ -0,0 +1,319 @@
+package controller
+
+import (
+	"encoding/json"
+	"errors"
+	"net"
+	"strconv"
+
+	"github.com/Microsoft/hcsshim/hcn"
+	"k8s.io/klog"
+
+	ps "github.com/bhendo/go-powershell"
+	psBackend "github.com/bhendo/go-powershell/backend"
+)
+
+// Datastore for NetworkInfo.
+type NetworkInfo struct {
+	ID           string
+	Name         string
+	Subnets      []SubnetInfo
+	Vsid         uint16
+	AutomaticDNS bool
+	IsPersistent bool
+}
+
+// Datastore for SubnetInfo.
+type SubnetInfo struct {
+	AddressPrefix  net.IPNet
+	GatewayAddress net.IP
+	Vsid           uint16
+}
+
+// GetHostComputeSubnetConfig converts SubnetInfo into an HCN format.
+func (subnet *SubnetInfo) GetHostComputeSubnetConfig() (*hcn.Subnet, error) {
+	// Check for nil on address objects.
+	ipAddr := ""
+	if subnet.AddressPrefix.IP != nil && subnet.AddressPrefix.Mask != nil {
+		ipAddr = subnet.AddressPrefix.String()
+	}
+	gwAddr := ""
+	destPrefix := ""
+	if subnet.GatewayAddress != nil {
+		gwAddr = subnet.GatewayAddress.String()
+		destPrefix = "0.0.0.0/0"
+	}
+
+	subnetPolicy := hcn.SubnetPolicy{
+		Type:     "VSID",
+		Settings: []byte("{ \"IsolationId\" : " + strconv.FormatUint(uint64(subnet.Vsid), 10) + " }"),
+	}
+
+	subnetPolicyJson, err := json.Marshal(subnetPolicy)
+
+	if err != nil {
+		klog.Error(err)
+		return nil, err
+	}
+
+	return &hcn.Subnet{
+		IpAddressPrefix: ipAddr,
+		Routes: []hcn.Route{{
+			NextHop:           gwAddr,
+			DestinationPrefix: destPrefix,
+		},
+		},
```

Comment:
I'm surprised gofmt tolerates this... it seems like it should be `}},`
(repeatedly though this file)